src/pool.c

```diff
@@ -1935,22 +1935,23 @@ rpc_on_wallet_transferred(const char* data, rpc_callback_t *callback)
         goto cleanup;
     }
     payment_t *payment = (payment_t*) callback->data;
-    for (; payment->amount; payment++)
+    if (payment)
```

Comment:
If there's no payment there is nothing to do. I.e. it would be an error getting into this function _without_ a payment. Thus if you're going to check for its presence, you may as well log an error and get out early, before even opening balance/payment transactions.